hub/util/path.py

```diff
@@ -0,0 +1,24 @@
+from hub.core.storage import LocalProvider
+import os
+
+
+def local_provider_from_path(path: str):
```

Comment:
is there a reason this method only supports a local provider?

Reply:
i recommend doing `provider_from_path` and just go through all of the available providers. maybe add a static abstract method to the provider base like `is_valid_path` that way you can just go through each of the providers here and if one matches, go with it